src/Fixer/ReturnNotation/ReturnAssignmentFixer.php

```diff
@@ -354,4 +373,98 @@ private function clearIfSave(Tokens $tokens, int $index): void
 
         $tokens->clearTokenAndMergeSurroundingWhitespace($index);
     }
+
+    /**
+     * @param int $index open brace index
+     *
+     * @return null|int index of the first token of a definition (lambda, anonymous class or match) or `null` if not an anonymous
+     */
+    private function isCloseBracePartOfDefinition(Tokens $tokens, int $index): ?int
+    {
+        $index = $tokens->findBlockStart(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+        $candidateIndex = $this->isOpenBraceOfLambda($tokens, $index);
+
+        if (null !== $candidateIndex) {
+            return $candidateIndex;
+        }
+
+        $candidateIndex = $this->isOpenBraceOfAnonymousClass($tokens, $index);
+
+        return $candidateIndex ?? $this->isOpenBraceOfMatch($tokens, $index);
+    }
+
+    /**
+     * @param int $index open brace index
+     *
+     * @return null|int index of T_NEW of anonymous class or `null` if not an anonymous
+     */
+    private function isOpenBraceOfAnonymousClass(Tokens $tokens, int $index): ?int
+    {
+        $index = $tokens->getPrevMeaningfulToken($index);
+
+        if ($tokens[$index]->equals(')')) {
+            $index = $tokens->findBlockStart(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+            $index = $tokens->getPrevMeaningfulToken($index);
+        }
+
+        if (!$tokens[$index]->isGivenKind(T_CLASS)) {
+            return null;
+        }
+
+        $index = $tokens->getPrevMeaningfulToken($index);
+
+        return $tokens[$index]->isGivenKind(T_NEW) ? $index : null;
+    }
+
+    /**
+     * @param int $index open brace index
+     *
+     * @return null|int index of T_FUNCTION or T_STATIC of lambda or `null` if not a lambda
+     */
+    private function isOpenBraceOfLambda(Tokens $tokens, int $index): ?int
+    {
+        $index = $tokens->getPrevMeaningfulToken($index);
+
+        if (!$tokens[$index]->equals(')')) {
+            return null;
+        }
+
+        $index = $tokens->findBlockStart(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+        $index = $tokens->getPrevMeaningfulToken($index);
+
+        if ($tokens[$index]->isGivenKind(CT::T_USE_LAMBDA)) {
+            $index = $tokens->getPrevTokenOfKind($index, [')']);
+            $index = $tokens->findBlockStart(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+            $index = $tokens->getPrevMeaningfulToken($index);
+        } elseif (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
```

Comment:
Shouldn't we check `T_FN` as well?

Reply:
These are already supported, the issues resolved in the PR are around `static` and the `&` for nested functions.
I've added a test for an arrow function as it was missing.
:+1: